packages/frontend/core/src/modules/peek-view/view/peek-view-manager.tsx

```diff
@@ -0,0 +1,136 @@
+import {
+  AffineReference,
+  type SurfaceRefBlockComponent,
+} from '@blocksuite/blocks';
+import { type DocMode, useLiveData, useService } from '@toeverything/infra';
+import { useMemo } from 'react';
+
+import type { ActivePeekView } from '../entities/peek-view';
+import { PeekViewService } from '../services/peek-view';
+import { DocPeekViewControls } from './doc-peek-controls';
+import { DocPeekView, SurfaceRefPeekView } from './doc-peek-view';
+import { PeekViewModalContainer } from './modal-container';
+import { isEmbedDocModel, isSurfaceRefModel, resolveLinkToDoc } from './utils';
+
+type DocPeekViewInfo = {
+  docId: string;
+  blockId?: string;
+  mode?: DocMode;
+  xywh?: `[${number},${number},${number},${number}]`;
+};
+
+function resolveDocIdFromPeekTarget(
+  peekTarget?: ActivePeekView['target']
```

Comment:
Passing the `HTMLElement` feels a bit dirty, perhaps passing the parameter directly would be more maintainable.

Reply:
```
peekView.open({
    docId: 'xxx'
})
```

Rather than

```
peekView.open(aElement)
```

Reply:
Passing DOM elements because we may need to animate on that element - like calculating the peeking screen position. That's why I retained the htmlelement in the target